rest/route/agent.go

```diff
@@ -922,37 +929,42 @@ func (h *startTaskHandler) Run(ctx context.Context) gimlet.Responder {
 		event.LogBuildStateChangeEvent(t.BuildId, updates.BuildNewStatus)
 	}
 
-	host, err := host.FindOne(host.ByRunningTaskId(t.Id))
-	if err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding host running task %s", t.Id))
-	}
-
-	if host == nil {
-		message := fmt.Sprintf("no host found running task %s", t.Id)
-		if t.HostId != "" {
-			message = fmt.Sprintf("no host found running task %s but task is said to be running on %s",
-				t.Id, t.HostId)
+	var msg string
+	if h.hostID != "" {
+		host, err := host.FindOne(host.ByRunningTaskId(t.Id))
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding host running task %s", t.Id))
 		}
 
-		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-			StatusCode: http.StatusNotFound,
-			Message:    message,
-		})
-	}
+		if host == nil {
+			message := fmt.Sprintf("no host found running task %s", t.Id)
+			if t.HostId != "" {
+				message = fmt.Sprintf("no host found running task %s but task is said to be running on %s",
+					t.Id, t.HostId)
+			}
 
-	idleTimeStartAt := host.LastTaskCompletedTime
-	if idleTimeStartAt.IsZero() || idleTimeStartAt == utility.ZeroTime {
-		idleTimeStartAt = host.StartTime
-	}
+			return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+				StatusCode: http.StatusNotFound,
+				Message:    message,
+			})
+		}
+
+		idleTimeStartAt := host.LastTaskCompletedTime
+		if idleTimeStartAt.IsZero() || idleTimeStartAt == utility.ZeroTime {
+			idleTimeStartAt = host.StartTime
+		}
 
-	msg := fmt.Sprintf("Task %s started on host %s", t.Id, host.Id)
+		msg = fmt.Sprintf("task %s started on host %s", t.Id, host.Id)
 
-	if host.Distro.IsEphemeral() {
-		queue := h.env.RemoteQueue()
-		job := units.NewCollectHostIdleDataJob(host, t, idleTimeStartAt, t.StartTime)
-		if err = queue.Put(ctx, job); err != nil {
-			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "queuing host idle stats for %s", msg))
+		if host.Distro.IsEphemeral() {
+			queue := h.env.RemoteQueue()
+			job := units.NewCollectHostIdleDataJob(host, t, idleTimeStartAt, t.StartTime)
+			if err = queue.Put(ctx, job); err != nil {
+				return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "queuing host idle stats for %s", msg))
+			}
 		}
+	} else {
+		msg = fmt.Sprintf("task %s started on container %s", t.Id, t.Container)
```

Comment:
Since there's no equivalent to the host idle data job, can we add a TODO here + ticket to eventually collect these stats? Would be good to know how long the pod is idle before running the task.

Reply:
Sounds good, created: https://jira.mongodb.org/browse/EVG-17647

Shall we add this to a specific epic or leave it to triage?

Reply:
I think we can leave it for triage and do it whenever there's time. There's already a couple of small one-off container tickets which aren't that urgent or relevant to any epic.